ksql-engine/src/main/java/io/confluent/ksql/util/PersistentQueryMetadata.java

```diff
@@ -134,6 +137,11 @@ public PhysicalSchema getPhysicalSchema() {
       final QueryId queryId,
       final QueryContext.Stacker contextStacker
   ) {
-    return materializationProvider.map(builder -> builder.build(queryId, contextStacker));
+    if (!materialization.isPresent()) {
```

Comment:
I don't think we should be doing lazy creation of this stuff.  We should be building it when we build the aggregate query.  This will simplify the code, i.e. we won't have both `materializationProvider` and `materialization`, but only `materialization`.

Oh, and there's also a potential race condition here, though I'm not sure it's got any negative implications other than potentially excessive creation of multiple materializations. Still, I'd go with `AtomicReference` rather than a mutable `Optional` for this kind of pattern.

Reply:
Good catch! Will fix